.github/workflows/retrieve_cache.yml

```diff
@@ -27,43 +27,53 @@
 #          key: ${{ needs.stpsf_data_cache_key.outputs.cache_key }}
 #       ...
 
-name: retrieve latest data cache key
+name: retrieve cache
 
 on:
   workflow_call:
     inputs:
-      minimal:
-        description: minimal dataset
-        type: boolean
+      dataset_type:
+        description: "Choose dataset type ('development' or 'latest_release')"
+        type: string
         required: false
-        default: true
+        default: development
     outputs:
-      version:
-        value: ${{ jobs.retrieve_latest_cache_key.outputs.version }}
       cache_path:
-        value: ${{ jobs.retrieve_latest_cache_key.outputs.cache_path }}
+        value: ${{ jobs.retrieve.outputs.cache_path }}
       cache_key:
-        value: ${{ jobs.retrieve_latest_cache_key.outputs.cache_key }}
+        value: ${{ jobs.retrieve.outputs.cache_key }}
+      version:
+        value: ${{ jobs.retrieve.outputs.version }}
 
 jobs:
-  retrieve_latest_cache_key:
-    name: retrieve latest STPSF data cache key
+  retrieve:
     runs-on: ubuntu-latest
+    outputs:
+      cache_path: ${{ steps.set-outputs.outputs.cache_path }}
+      cache_key: ${{ steps.set-outputs.outputs.cache_key }}
+      version: ${{ steps.set-outputs.outputs.version }}
     steps:
-      - name: retrieve latest data cache key
-        id: latest_cache_key
+      - name: Determine cache key
+        id: cache-key
         run: |
-          # use actions/gh-actions-cache to allow filtering by key
-          gh extension install actions/gh-actions-cache
-          CACHE_KEY=$(gh actions-cache list -R ${{ github.repository }} --key stpsf-data-${{ inputs.minimal && 'mini' || 'full' }}- --sort created-at | cut -f 1 | head -n 1)
-          if [ "$CACHE_KEY" == '' ]; then exit 1; fi
-          echo cache_key=$CACHE_KEY >> $GITHUB_OUTPUT
-        env:
-          GH_TOKEN: ${{ github.token }}
-      - run: echo ${{ steps.latest_cache_key.outputs.cache_key }}
-      - id: version
-        run: echo "version=$(echo ${{ steps.latest_cache_key.outputs.cache_key }} | awk -F '-' '{print $4}')" >> $GITHUB_OUTPUT
-    outputs:
-      version: ${{ steps.version.outputs.version }}
-      cache_path: ${{ runner.temp }}/data/
-      cache_key: ${{ steps.latest_cache_key.outputs.cache_key }}
+          PREFIX="stpsf-data-${{ inputs.dataset_type }}-"
+
+          # Find the most recent cache key matching dataset type
+          CACHE_KEY=$(gh actions-cache list -R ${{ github.repository }} --key "$PREFIX" --sort created-at | cut -f1 | head -n1)
+
+          echo "cache_key=$CACHE_KEY" >> $GITHUB_OUTPUT
+
+          # Extract version (suffix after prefix, if present)
+          VERSION=${CACHE_KEY#"$PREFIX"}
```

Comment:
how does this work?

Reply:
this is a bash syntax that will strip $PREFIX (defined above) from the CACHE_KEY leaving only the version which is saved as VERSION.  You can try it in terminal :) 

`$ PREFIX="stpsf-data-development-"
$ TEMP_KEY="stpsf-data-development-20250818"
$ echo ${TEMP_KEY#"$PREFIX"}`